cmake/podioMacros.cmake

```diff
@@ -121,6 +121,18 @@ function(PODIO_GENERATE_DATAMODEL datamodel YAML_FILE RETURN_HEADERS RETURN_SOUR
   IF(NOT ARG_OUTPUT_FOLDER)
     SET(ARG_OUTPUT_FOLDER ${CMAKE_CURRENT_SOURCE_DIR})
   ENDIF()
+
+  # Make sure that no remnants from previous runs are still present in the
+  # output directory, to avoid running into unexpected problems with "old"
+  # generation files
+  IF(NOT __STARTED_GENERATION_FROM_CLEAN_SLATE)
```

Comment:
Does this really work as intended?

1.) cmake .. # cleans and sets __STARTED... True
2.) git pull  # e.g.,
3.) cmake .. # doesn't clean

Right?

Reply:
Yes, you are right. However, in that case you should still have a "valid" build file and can do something along the lines of `make clean` to force the cleanup of the generator output directory. This fix should make starting in a clean build directory more predictable. E.g.
```bash
mkdir build && cd build
cmake ..
```
Then do whatever needs to be done here. Suppose you then want to start from a clean slate, but you forgot to `make clean` before doing the usual `rm -rf build` and then start the process from the beginning. In this case it is likely that the generated files from the previous run are still around and possibly pollute the new run, e.g. when some files are no longer generated, because the class has been removed from the yaml file in between.

Reply:
However, maybe it would be more intuitive if each call to `cmake` cleans up the generator output directories? This would cover my intended use case for free basically.

Stupid question: Everything in `PODIO_GENERATE_DATAMODEL` is only run when `cmake` is invoked, right? The possible regeneration of the datamodel via a call to `make` is handled by the custom target that we add? In that case I think it would be possible to simply remove the generator output directories unconditionally without the usage of a cache variable.

Reply:
> I would be using `git clean` at this point. 

Ah I hadn't thought about that possibility in fact. But that would probably also clean a lot of things it is not supposed to, e.g. files that are currently being worked on but not yet tracked.

> Or the podio output folder should go to CMAKE_BINARY_DIR , a.k.a. the "build" folder.

The problem with that is that it might be desirable to have the generated code somewhere else, e.g. for documentation purposes, or possibly to put it under version control (for some reason).

> If one wants no-longer needed generated files to be removed (or ignored), maybe it would be better for podio to spit out the created files instead of using the cmake globbing of files?

Yes, that is probably the most robust approach. It would also allow users to put other files into the generator output directories if they desire to do so.